typescript/schedules.md

```diff
@@ -0,0 +1,161 @@
+# Schedules
+
+- [Docs](https://docs.temporal.io/workflows/#schedules)
+- gRPC API:
+  - [methods](https://github.com/temporalio/api/blob/799926c86eb13d8a9717d3561ab9b0df43796c06/temporal/api/workflowservice/v1/service.proto#L328-L370)
+  - [`request_response.proto`](https://github.com/temporalio/api/blob/799926c86eb13d8a9717d3561ab9b0df43796c06/temporal/api/workflowservice/v1/request_response.proto#L821-L957)
+  - [`schedule/v1/message.proto`](https://github.com/temporalio/api/blob/master/temporal/api/schedule/v1/message.proto)
+  - [`ScheduleOverlapPolicy` enum](https://github.com/temporalio/api/blob/master/temporal/api/enums/v1/schedule.proto)
+
+## TS API
+
+```ts
+import { ScheduleClient, ScheduleOverlapPolicy } from '@temporalio/client'
+import { addWeeks } from 'date-fns'
+import { myWorkflow } from './workflows'
+
+const client = new ScheduleClient()
+
+const schedule = await client.create({
+  id: 'schedule-biz-id',
+  spec: {
+    // every hour at minute 5
+    interval: {
+      every: '1h',
+      at: '5m',
+    },
+    // every 20 days since epoch at day 2
+    // interval: {
+    //   every: '20d',
+    //   at: '2d'
+    // }
+    exclude: {
+      // skip 11:05 pm
+      hour: 23,
+      minute: 5,
+    },
+    endAt: addWeeks(new Date(), 4), 
+    jitter: '30s',
+    timezone: 'US/Eastern',
+  },
+  action: {
+    startWorkflow: {
+      workflowId: 'wf-biz-id',
+      type: myWorkflow,
+      args: ['sorry this is the only thing reused, chad 😄'],
+      // ...WorkflowOptions + header - followRuns
+      // https://typescript.temporal.io/api/interfaces/client.WorkflowOptions
+    },
+  },
+  policies: {
+    overlap: ScheduleOverlapPolicy.BUFFER_ONE,
+    catchupWindow: '2m',
+    pauseOnFailure: true,
+  },
+  state: {
+    note: 'started schedule',
+    paused: true,
+    limitedActions: 10,
+  },
+  patch: {
+    triggerImmediately: true,
+    backfill: [
+      {
+        start: new Date(),
+        end: new Date(),
+        overlap: ScheduleOverlapPolicy.ALLOW_ALL,
+      },
+    ],
+    pause: true, // redundant with state.paused above (can use either)
+  },
+  memo,
+  searchAttributes,
+})
+
+const schedule = client.getHandle('schedule-biz-id')
+
+const scheduleDescription = await schedule.describe()
+
+const matchingStartTimes = await schedule.listMatchingTimes({ start: new Date(), end: new Date() })
```

Comment:
this isn't really a "listing api", it's a query to an individual schedule to just evaluate its spec over a range of times. I think the more useful query would accept a full spec and range of times and evaluate it. I could add that as a new rpc except that it's completely stateless so a little weird.

Reply:
I see. Yeah that is a bit weird to ask the server to give you schedule info you _could_ calculate yourself. But even if stateless, seems like it could have value as that kind of utility is hard to write accurately.

But I vote for leaving it out of the SDK API for now if we're not sure it's the way we want it.

Reply:
Even if current method is replaced by the stateless version that takes a spec, it seems nice/useful to have a method on a handle. Like:

```ts
client.listMatchingTimes({ spec, start, end })
scheduleHandle.listMatchingTimes({ start, end })
```

where the second one internally calls `describe` to get the spec and then `client.listMatchingTimes`.

Reply:
Let's just leave this off for now

Reply:
I'd like it even if no API and logic is implemented clientside, but can leave commented out for now.